MastodonSDK/Sources/MastodonUI/View/Window/TouchesVisibleWindow.swift

```diff
@@ -0,0 +1,136 @@
+        }
+    }
+    
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        
+        let isLightMode = traitCollection.userInterfaceStyle == .light
+        
+        backgroundColor = .clear
+        layer.masksToBounds = true
+        layer.cornerCurve = .circular
+        layer.borderColor = isLightMode ? UIColor.gray.cgColor : UIColor.white.cgColor
+        layer.borderWidth = 2.0
+        
+        let blurEffect = isLightMode ?
+            UIBlurEffect(style: .systemUltraThinMaterialDark) :
+            UIBlurEffect(style: .systemUltraThinMaterialLight)
+        blurView.effect = blurEffect
+        addSubview(blurView)
+    }
+    
+    @available(iOS, unavailable)
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    override func layoutSubviews() {
+        super.layoutSubviews()
+        blurView.frame = bounds
+    }
+    
+}
+
+
+/// `UIWindow` subclass that renders visual representations of the user's touches.
+public final class TouchesVisibleWindow: UIWindow {
+    
+    public var touchesVisible = false {
+        didSet {
+            if !touchesVisible {
+                cleanUpAllTouches()
+            }
+        }
+    }
+    
+    private var touchViews: [UITouch : TouchView] = [:]
+    
+    private func newTouchView() -> TouchView {
+        let touchSize = 44.0
+        return TouchView(frame: CGRect(
+            origin: .zero,
+            size: CGSize(
+                width: touchSize,
+                height: touchSize
+            )
+        ))
+    }
+    
+    private func cleanupTouch(_ touch: UITouch) {
+        guard let touchView = touchViews[touch] else {
+            return
+        }
+        
+        touchView.removeFromSuperview()
+        touchViews.removeValue(forKey: touch)
+    }
+    
+    private func cleanUpAllTouches() {
+        for (_, touchView) in touchViews {
+            touchView.removeFromSuperview()
+        }
+        
+        touchViews.removeAll()
+    }
+    
+    public override func sendEvent(_ event: UIEvent) {
+        if touchesVisible {
+            let touches = event.allTouches
+            
+            guard
+                let touches = touches,
+                touches.count > 0
+            else {
+                cleanUpAllTouches()
+                super.sendEvent(event)
+                return
+            }
+            
+            for touch in touches {
+                let touchLocation = touch.location(in: self)
+                switch touch.phase {
+                case .began:
+                    let touchView = newTouchView()
+                    touchView.center = touchLocation
+                    addSubview(touchView)
+                    touchViews[touch] = touchView
+                    
+                case .moved:
+                    guard let touchView = touchViews[touch] else {
+                        return
+                    }
```

Comment:
just curious, do we not require to call `super.sendEvent(event)` here?

Reply:
Good catch! There does need to be super call in this case. 

On second look, this early return actually feels kinda odd. I'll update this conditional to be an if-let, and we'll naturally fall through to the super call at the end of the function.